File: packages/sb-uhq/src/self/commands/fun/mock.js
"use strict";

const { messageEdit } = require("../../func/message_edit");

// ── Logique pure ─────────────────────────────────────────────────────────────

function mockText(text) {
  return text.split("").map((c, i) => i % 2 === 0 ? c.toLowerCase() : c.toUpperCase()).join("");
}

/**
 * @param {import("discord.js-selfbot-v13").Client} client
 * @param {{ channelId: string, text: string }} payload
 */
async function execute(client, payload) {
  const { channelId, text } = payload;
  if (!channelId || !text) throw new Error("channelId et text requis.");

  const channel = await client.channels.fetch(channelId).catch(() => null);
  if (!channel) throw new Error(`Salon ${channelId} introuvable.`);

  const mocked = mockText(text);
  await channel.send(mocked);
  return { sent: mocked };
}

// ── Handler commande préfixe ──────────────────────────────────────────────────

async function callback(client, message, args) {
  if (!args.length) return messageEdit(message, "`❌` **Usage :** `.mock <texte>`");
  const mocked = mockText(args.join(" "));
  return message.edit(mocked).catch(() => message.channel.send(mocked));
}

module.exports = { name: "mock", execute, callback };

File: packages/sb-uhq/app/commands/fun/mock.py
"""fun.mock — tExTe En MoDe MoCk (bridge + commande préfixe)."""

from __future__ import annotations

from ...func.discord_util import fetch_channel
from ...func.message_edit import message_edit


def mock_text(text: str) -> str:
    return "".join(c.lower() if i % 2 == 0 else c.upper() for i, c in enumerate(text))


async def execute(client, payload):
    channel_id = payload.get("channelId")
    text = payload.get("text")
    if not channel_id or not text:
        raise ValueError("channelId et text requis.")
    channel = await fetch_channel(client, channel_id)
    if not channel:
        raise ValueError(f"Salon {channel_id} introuvable.")
    mocked = mock_text(text)
    await channel.send(mocked)
    return {"sent": mocked}


async def callback(client, message, args):
    if not args:
        return await message_edit(message, "`❌` **Usage :** `.mock <texte>`")
    mocked = mock_text(" ".join(args))
    return await message_edit(message, mocked)
